zetaclient/zetacore/client.go

```diff
@@ -367,87 +362,72 @@ func (c *Client) UpdateAppContext(
-	var newSolanaParams *observertypes.ChainParams
-
-	// check and update chain params for each chain
-	for _, chainParam := range chainParams {
-		err := observertypes.ValidateChainParams(chainParam)
-		if err != nil {
-			sampledLogger.Warn().Err(err).Msgf("Invalid chain params for chain %d", chainParam.ChainId)
-			continue
-		}
-		if chains.IsBitcoinChain(chainParam.ChainId, additionalChains) {
-			newBTCParams = chainParam
-		} else if chains.IsSolanaChain(chainParam.ChainId, additionalChains) {
-			newSolanaParams = chainParam
-		} else if chains.IsEVMChain(chainParam.ChainId, additionalChains) {
-			newEVMParams[chainParam.ChainId] = chainParam
-		}
-	}
-
-	supportedChains, err := c.GetSupportedChains(ctx)
+	additionalChains, err := c.GetAdditionalChains(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to get supported chains: %w", err)
+		return errors.Wrap(err, "unable to fetch additional chains")
 	}
 
-	newChains := make([]chains.Chain, len(supportedChains))
-	for i, chain := range supportedChains {
-		newChains[i] = chain
+	chainParams, err := c.GetChainParams(ctx)
+	if err != nil {
+		return errors.Wrap(err, "unable to fetch chain params")
 	}
 
 	keyGen, err := c.GetKeyGen(ctx)
 	if err != nil {
-		c.logger.Info().Msg("Unable to fetch keygen from zetacore")
-		return fmt.Errorf("failed to get keygen: %w", err)
+		return errors.Wrap(err, "unable to fetch keygen from zetacore")
 	}
 
-	tss, err := c.GetCurrentTSS(ctx)
+	crosschainFlags, err := c.GetCrosschainFlags(ctx)
 	if err != nil {
-		c.logger.Info().Err(err).Msg("Unable to fetch TSS from zetacore")
-		return fmt.Errorf("failed to get current tss: %w", err)
+		return errors.Wrap(err, "unable to fetch crosschain flags from zetacore")
 	}
-	tssPubKey := tss.GetTssPubkey()
 
-	crosschainFlags, err := c.GetCrosschainFlags(ctx)
+	tss, err := c.GetCurrentTSS(ctx)
 	if err != nil {
-		c.logger.Info().Msg("Unable to fetch cross-chain flags from zetacore")
-		return fmt.Errorf("failed to get crosschain flags: %w", err)
+		return errors.Wrap(err, "unable to fetch current TSS")
 	}
 
-	blockHeaderEnabledChains, err := c.GetBlockHeaderEnabledChains(ctx)
-	if err != nil {
-		c.logger.Info().Msg("Unable to fetch block header enabled chains from zetacore")
-		return err
+	freshParams := make(map[int64]*observertypes.ChainParams, len(chainParams))
+
+	// check and update chain params for each chain
+	// Note that we are EXCLUDING ZetaChain from the chainParams if it's present
+	for i := range chainParams {
+		cp := chainParams[i]
+
+		if !cp.IsSupported {
+			logger.Warn().Int64("chain.id", cp.ChainId).Msg("Skipping unsupported chain")
+			continue
+		}
+
+		if chains.IsZetaChain(cp.ChainId, nil) {
+			continue
+		}
+
+		if err := observertypes.ValidateChainParams(cp); err != nil {
+			logger.Warn().Err(err).Int64("chain.id", cp.ChainId).Msg("Skipping invalid chain params")
+			continue
+		}
+
+		freshParams[cp.ChainId] = cp
 	}
 
-	appContext.Update(
+	return appContext.Update(
```

Comment:
Can we try skipping the chain params, supported chains, and additional chains if they don't change? 
The frequency of change for these values is quite less

Reply:
Okay, I think that's fine.
However, the list of chains can increase to a much higher number in the future, which might increase this write time, and having some other threads using these values might add additional complexities. 

feel free to resolve this for now and create an issue if you think it would need a refactor later if we start supporting a lot more chains